opensearch/src/main/java/org/opensearch/sql/opensearch/data/type/OpenSearchDataType.java

```diff
@@ -230,6 +234,9 @@ public String legacyTypeName() {
     if (mappingType == null) {
       return exprCoreType.typeName();
     }
+    if (mappingType.toString().equalsIgnoreCase("DATE")) {
+      return exprCoreType.typeName();
+    }
```

Comment:
I think we cannot move this check from `legacyTypeName()` to `typeName()` function as it will fail IT tests. I think `legacyTypeName` is used to serialize SQL response on which few IT tests depends on and `typeName` is used to map and serialize PPL response.